test/functional/launch/shared_mem/tests/test-launch-DynamicMem.hpp

```diff
@@ -0,0 +1,111 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-22, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#ifndef __TEST_LAUNCH_DYNAMIC_MEM_HPP__
+#define __TEST_LAUNCH_DYNAMIC_MEM_HPP__
+
+#include <numeric>
+
+template <typename INDEX_TYPE, typename WORKING_RES, typename LAUNCH_POLICY, typename TEAM_POLICY, typename THREAD_POLICY>
+void LaunchDynamicMemTestImpl(INDEX_TYPE block_range, INDEX_TYPE thread_range)
+{
+
+  RAJA::TypedRangeSegment<INDEX_TYPE> outer_range(RAJA::stripIndexType(INDEX_TYPE(0)), RAJA::stripIndexType(block_range));
+  RAJA::TypedRangeSegment<INDEX_TYPE> inner_range(RAJA::stripIndexType(INDEX_TYPE(0)), RAJA::stripIndexType(thread_range));
+
+  camp::resources::Resource working_res{WORKING_RES::get_default()};
+  INDEX_TYPE* working_array;
+  INDEX_TYPE* check_array;
+  INDEX_TYPE* test_array;
+  
+  size_t data_len = RAJA::stripIndexType(block_range)*RAJA::stripIndexType(thread_range);
+  
+  allocateForallTestData<INDEX_TYPE>(data_len,
+                                     working_res,
+                                     &working_array,
+                                     &check_array,
+                                     &test_array);
+  
+  
+  for(int b=0; b<RAJA::stripIndexType(block_range); ++b) {
+    for(int c=0; c<RAJA::stripIndexType(thread_range); ++c) {
+      int idx = c + RAJA::stripIndexType(thread_range)*b;
+      test_array[idx] = INDEX_TYPE(b);
```

Comment:
Would it make more sense to set them to idx instead of b so they are each unique?

Reply:
Well the test I had in mind was to do a broadcast with shared memory. For example in shared memory store the block number (that would use an array of size one) and then each thread can read from it. That would test that the threads can indeed read from a shared location unique to each block (so threads from different blocks would be reading a different value).

Reply:
Just noticed what was pushed was not what I had in mind, just pushed the correct implementation.

Reply:
That is a good idea to actually read from memory set by a different thread. How about you use idx, but set it in a different thread?

Reply:
Done, I used your idea from below.